build/distributions/CentOS/8/x86_64/DVD/ks/repodetect.sh

```diff
@@ -1,117 +1,114 @@
 #!/bin/sh
 #
-# Usage: repodetect.sh <Version> <yum_server>
+# repodetect.sh - determine kickstart repos based on your OS
+#
+# Usage: repodetect.sh VERSION [KS_SERVER] [LINUX_DIST]
+#
+#   VERSION     Major OS version number (e.g., '8', '7', '6')
+#   YUM_SERVER  (Optional) empty, 'local', or a URI of a yum server
+#   LINUX_DIST  (Optional) Forces Linux  distro (e.g., 'CentOS', 'RedHat')
+#
+# Supported OSes: LINUX_DIST: [CentOS, RedHat],  VERSION: [6, 7, 8]
 #
-# Version: Mandatory
-# Yum_server:
-# - Blank or 'Local' -> use a local file source
-# - Anything else    -> use this is the URI at which to point.
 
 unknown="REPODETECT_UNKNOWN_OS_TYPE"
-# The following line does not work on EL8. The repos will default to
-# CentOS if the type is still unknown at the end of the script.
-osline=`dmesg -s 10485760 | grep '(Kernel Module GPG key)'`
-
-version=$1
-yum_server=$2
-
-if [ $# -gt 2 ]; then
-  type=$3
-else
-  type=$unknown
-fi
+distro="$unknown"
+arch="x86_64"
 
-if [ -z $version ]; then
-  echo "Error: You must pass <version> as the first argument";
-  exit 1;
-fi
+version="$1"
+yum_server="$2"
+if [ $# -gt 2 ]; then distro="$3"; fi
+if [ "$(uname -m)" != "$arch" ]; then arch="$(uname -m)"; fi
```

Comment:
why not just set `$arch` to `$(uname -m)` ?

Reply:
No reason at this point; changed.